doc: deprecate notice for _getActiveHandles and _getActiveRequests (#41587)
nodejs/node: doc/api/deprecations.md

Use [`buffer.subarray`][] which does the same thing instead.

> ### DEP0159: Deprecated `_getActiveRequestsInfo` and `_getActiveHandlesInfo`

Reviewer: ```suggestion
### DEP0159: `process._getActiveRequests()` and `process._getActiveHandles()`
```
The rest of the titles in this doc doesn't say `Deprecated` at the beginning because it's implied.

Author: Updated

<!-- YAML
changes:
  - version: REPLACEME
    pr-url: https://github.com/nodejs/node/pull/41587
    description: Documentation-only deprecation.
-->

Type: Documentation-only

> The `process._getActiveHandlesInfo()` and `process._getActiveRequestsInfo()`
> is not intended for public use and can be removed in future releases.

Reviewer: ```suggestion
The `process._getActiveHandles()` and `process._getActiveRequests()` functions
are not intended for public use and can be removed in future releases.
```

Author: Updated

> Use `process.getActiveResourcesInfo()` to get the list of handles and not
> the actual references.

Reviewer: ```suggestion
Use [`process.getActiveResourcesInfo()`][] to get a list of types of the active
resources instead of the actual references.
```
because resources = handles + requests here and you can add this line:
```md
[`process.getActiveResourcesInfo()`]: process.md#processgetactiveresourcesinfo 
```
after https://github.com/nodejs/node/blob/ccb8aae3932c13f33622203b2ffc5a33120e9d40/doc/api/deprecations.md#L3129, so that anyone can click the link and read more about the function.

Author: Updated

Use [`buffer.subarray`][] which does the same thing instead.

> ### DEP0159: `_getActiveRequests` and `_getActiveHandles`

Reviewer: ```suggestion
### DEP0159: `process._getActiveRequests()` and `process._getActiveHandles()`
```
Most of the other deprecation titles include the global object the deprecated functions belongs to. Also the `()` makes it clear that it's a function.

Author: Updated

> ### DEP0159: `process._getActiveRequests()` and `process._getActiveHandles()`

Reviewer 2: ```suggestion
### DEP0XXX: `process._getActiveRequests()` and `process._getActiveHandles()`
```

The deprecation number is assigned when the PR lands

Reviewer 3: Except that it's no longer the case since the CQ uses "Squash and merge" instead of pushing a different commit.

[`os.networkInterfaces()`]: os.md#osnetworkinterfaces
[`os.tmpdir()`]: os.md#ostmpdir
[`process.env`]: process.md#processenv

> [`process.getActiveResourcesInfo()`]: process.md#processenv

Reviewer: ```suggestion
[`process.getActiveResourcesInfo()`]: process.md#processgetactiveresourcesinfo
```

Author: Updated

> [`process.getActiveResourcesInfo()`]: process.md#getactiveresourcesinfo

Reviewer: ```suggestion
[`process.getActiveResourcesInfo()`]: process.md#processgetactiveresourcesinfo
```
This should fix the CI error :)

Author: Updated
